src/libertem/io/dataset/dm.py

```diff
@@ -125,6 +125,8 @@ def __init__(self, files=None, scan_size=None, same_offset=False, nav_shape=None
             self._nav_shape = tuple(scan_size)
         self._filesize = None
         self._files = files
+        assert isinstance(files, (list, tuple)),\
+               f"files argument must be an iterable of file paths recieved {type(files)}"
```

Comment:
An `assert` will be skipped when running Python in optimized mode. That means it can be used for internal consistency  checks. When validating user input, one should throw, for example, a `ValueError` instead.

Reply:
Understood, I've modified accordingly (`DataSetException` as that seems to be the style in the file).